lib/cli/etcher.js

```diff
@@ -18,62 +18,114 @@
 
 const _ = require('lodash');
 const Bluebird = require('bluebird');
+const inquirer = require('inquirer');
 const isElevated = Bluebird.promisify(require('is-elevated'));
 const visuals = require('resin-cli-visuals');
-const form = require('resin-cli-form');
 const drivelist = Bluebird.promisifyAll(require('drivelist'));
+const chalk = require('chalk');
+const imageStream = require('etcher-image-stream');
 const writer = require('./writer');
 const utils = require('./utils');
 const options = require('./cli');
+const DynamicList = require('./dynamic-list');
+const DriveScanner = require('./drive-scanner');
 const EXIT_CODES = require('../src/exit-codes');
 
-isElevated().then((elevated) => {
+inquirer.registerPrompt('dynamicList', DynamicList);
+
+const scanner = new DriveScanner(2000);
+
+const driveToChoice = (drive, minSizeToEnable) => {
+  const size = drive.size / 1000000000;
+
+  let displayName = `${drive.device} (${size.toFixed(1)} GB) - ${drive.description}`;
+
+  if (drive.system) {
+    displayName += chalk.red(' FIXED DRIVE');
+  }
+
+  const choice = {
+    name: displayName,
+    value: drive.device
+  };
+  if (drive.size < minSizeToEnable) {
+    choice.disabled = 'TOO SMALL FOR IMAGE';
+  }
+  return choice;
+};
+
+const mustConfirm = !options.robot && !options.yes;
+
+Bluebird.all([
+  isElevated(),
+  imageStream.getImageMetadata(options._[0])
+]).then((results) => {
```

Comment:
I would've loved to use rest parameters destruction here `then(...[elevated, imageMetadata')`, but `ESLint` complains.